src/PerformanceTests/Common/Scenarios/GatedPublish/GatedPublishRunner.cs

```diff
@@ -34,10 +34,7 @@ partial class GatedPublishRunner : LoopRunner, IConfigureUnicastBus
                     X.Reset(batchSize);
 
                     var d = Stopwatch.StartNew();
-
-                    var sends = new Task[X.InitialCount];
-                    for (var i = 0; i < X.InitialCount; i++) sends[i] = Publish(new Event());
-                    await Task.WhenAll(sends);
+                    await TaskHelper.ParallelFor(X.InitialCount, () => Session.Publish(new Event()));
```

Comment:
I understood that this is more for frameworks/libraries like NSB, not so much for your own code?